### PR TITLE
[codex] Support radial label rotation in chord chart

### LINKED ISSUE #21199: [Feature] can not set label.rotate to 'radial' in chord chart
### What problem does this feature solve?

The label of the current chord diagram is horizontal by default, but when the amount of data is large, label.rotate should be set to 'radial' so that the labels do not overlap.

Image

### What does the proposed API look like?

Add ‘radial’ option to label.rotate

### PR DESCRIPTION
**Pull request by susiwen8**
## Summary

- Add `label.rotate: 'radial'` support for chord node labels.
- Keep numeric chord `label.rotate` working by applying degree-to-radian conversion in the chord label layout path.
- Add a regression test that verifies rendered chord node text receives the expected radial rotation.

Fixes #21199.

## Validation

- `npx jest --config test/ut/jest.config.cjs --runTestsByPath test/ut/spec/series/chord.test.ts --coverage=false`
- `npm test -- --runInBand --coverage=false`
- `npm run checktype`
- `npm run lint -- --no-cache src/chart/chord/ChordPiece.ts src/chart/chord/ChordSeries.ts`
- `npx eslint --no-cache test/ut/spec/series/chord.test.ts`
- `npx tsc -p test/ut/tsconfig.json --noEmit --moduleResolution node --skipLibCheck`

## Note

`npm run checkheader` currently fails on pre-existing files outside this change (`test/ut/coverage/*`, `test/line-area-empty-dimension-name.html`, and `test/visualmap-seriesTargets.html`). The files changed in this PR have license headers.

## ScreenShot
image

### CODE REVIEW

File: src/chart/chord/ChordSeries.ts [hunk 1]
```diff
@@ -97,9 +97,10 @@ export interface ChordEdgeLineStyleOption extends LineStyleOption {
     curveness?: number
 }
 
-export interface ChordNodeLabelOption extends Omit<SeriesLabelOption<CallbackDataParams>, 'position'> {
+export interface ChordNodeLabelOption extends Omit<SeriesLabelOption<CallbackDataParams>, 'position' | 'rotate'> {
     silent?: boolean
     position?: SeriesLabelOption['position'] | 'outside'
+    rotate?: number | 'radial'
```
[the remark this answers is not in the document]
Comment (susiwen8) [reply src/chart/chord/ChordSeries.ts:103]: Addressed in b022a719f by using ChordNodeLabelOption for chord state labels, so emphasis/blur/select labels accept rotate: radial consistently.

File: test/ut/spec/series/chord.test.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    it('supports radial label rotation', function () {
+        chart.setOption({
+            series: {
+                type: 'chord',
+                startAngle: 0,
+                padAngle: 0,
+                label: {
+                    show: true,
+                    rotate: 'radial'
+                },
```
[the remark this answers is not in the document]
Comment (susiwen8) [reply test/ut/spec/series/chord.test.ts:67]: Addressed in b022a719f by adding a numeric rotation regression case that verifies 45 degrees maps to Math.PI / 4.

File: src/chart/chord/ChordPiece.ts [hunk 3]
```diff
@@ -26,9 +26,18 @@ import type Model from '../../model/Model';
...
 import type { BuiltinTextPosition } from 'zrender/src/core/types';
-import { setStatesStylesFromModel, toggleHoverEmphasis } from '../../util/states';
+import { setStatesStylesFromModel, SPECIAL_STATES, toggleHoverEmphasis } from '../../util/states';
 import { getECData } from '../../util/innerStore';
 
+const RADIAN = Math.PI / 180;
+
+function getLabelRotation(rotate: number | 'radial', midAngle: number, dx: number): number {
+    if (rotate === 'radial') {
+        return dx < 0 ? -midAngle + Math.PI : -midAngle;
```
Comment (Ovilia) [inline src/chart/chord/ChordPiece.ts]: Does `series.startAngle` not being default value affect the result?

**Review by Ovilia — COMMENTED** [review]
You may consider referring to sunburst's label rotate for boundary considerations.

File: src/chart/chord/ChordPiece.ts (hunk 3 above)
Comment (susiwen8) [reply src/chart/chord/ChordPiece.ts]: `series.startAngle` is already included in the node layout: `chordLayout` derives each nodes `layout.startAngle` / `layout.endAngle` from the normalized series start angle, and the label rotation uses that resulting `midAngle`.

The boundary consideration is valid, though. The current `dx < 0` flip check can be sensitive around the 90 / 270 degree boundaries, especially with non-default `startAngle`, while sunburst uses normalized radians plus an epsilon boundary check. I will align this with the sunburst approach and add coverage for a non-default `startAngle` boundary case.

Comment (susiwen8) [reply src/chart/chord/ChordPiece.ts]: Updated in b45a9b7: chord radial label rotation now uses normalized-angle boundary logic aligned with sunburst, and the unit test covers a non-default `startAngle` boundary case at 270 degrees.

`startAngle` is already applied by `chordLayout` before `ChordPiece` receives the node `midAngle`, so this keeps the existing startAngle behavior while avoiding `dx < 0` precision-sensitive flips around the vertical boundaries.
